object/src/mem.rs

```diff
@@ -684,6 +684,10 @@ where
         self.entries.get(&tag)
     }
 
+    pub fn get_mut(&mut self, tag: Tag) -> Option<&mut InMemElement<D>> {
```

Comment:
This one is a no-go, because that makes it possible to change the tag and consequently diverge from the dictionary key.

If `update_value` is too limiting, the safe approach to modifying an entire data element is to remove it and then insert it back.